src/AssignmentThree.java

```diff
@@ -182,12 +220,101 @@ public void setErrorFlag(boolean errorFlag)
+ * we must keep track of this with an int value (numCards).  
+ * 
+ * We'll need constructors, mutators, etc., of course.  
+ * 
+ * We'll also want a way for the hand to receive a card (from the deck or somewhere else), 
+ * 
+ * and play a card (to the table or to another player).  
+ * 
+ * These two methods will be called takeCard() and playCard(), respectively. 
+ * 
+ * Since this class has no information about the game being played, it always puts new cards received by takeCard() into the next available location of the array (index position numCards) 
+ * 
+ * and plays a card via playCard() from the highest occupied location (index position numCards - 1).  
+ * 
+ * The client game application would somehow prepare this highest position with the correct card to be played before calling Hand's playCard() method. 
+ * 
+ *  This detail is not our concern.
+ * @author charlesk
+ *
  */
 class Hand
 {
-   // empty
+    public static final int MAX_CARDS = 100;
+    private Card[] myCards;
+    private int numCards;
+    
+    public Hand() {
+        //create a default array of Cards
+        this.myCards = new Card[MAX_CARDS];
+    }
+    
+    /**
+     * remove all cards from the hand (in the simplest way).
+     */
+    public void resetHand() {
+        //nothing to reset as myCards is either null or it's reset already.
+        if (null == this.myCards || this.myCards.length == 0) {
+            return;
+        } 
+        this.myCards = new Card[MAX_CARDS];
+    }
+    
+    /**
+     * adds a card to the next available position in the myCards array.
+     * This is an object copy, not a reference copy, since the source of the Card might destroy or change 
+     * its data after our Hand gets it -- we want our local data to be exactly as it was when we received it.
+     * @param card
+     * @return true if card was successfully added to the myCards array, otherwise return false.
+     */
+    public boolean takeCard(Card card) {
+        if (this.numCards == this.myCards.length) {
+            return false;
+        }
+        Card cardCopy = Card.newInstance(card);
+        this.myCards[numCards++] = cardCopy;
+        return true;
+    }
+    
+    /**
+     * 
+     * @return Card and removes the card in the top occupied position of the array.
+     */
+    public Card playCard() {
+        Card card = this.myCards[--numCards];
+        this.myCards[numCards] = null;
+        return card; 
+    }
+
+    /**
+     * 
+     */
+    @Override
+    public String toString() {
+        return "Hand [myCards=" + Arrays.toString(myCards) + ", numCards=" + numCards + "]";
+    }
+    
+    /**
+     * Accessor for an individual card.
+     * 
+     * @param k
+     * @return a card with errorFlag = true if k is bad.
+     */
+    public Card inspectCard(int k) {
+        if (null == this.myCards || this.myCards.length == 0 || k > this.myCards.length) {
```

Comment:
Consider flipping the check from null == this.myCards to this.myCards == null to be consistent (there are a few places of this)